app/v2/services/MessagesXmlParsingService.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.services
+
+import akka.stream.Materializer
+import akka.stream.alpakka.xml.StartElement
+import akka.stream.alpakka.xml.scaladsl.XmlParsing
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import cats.data.EitherT
+import com.google.inject.ImplementedBy
+import com.google.inject.Inject
+import v2.models.errors.FailedToValidateError
+import v2.models.request.MessageType
+
+import javax.inject.Singleton
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[MessagesXmlParsingServiceImpl])
+trait MessagesXmlParsingService {
+  def validateMessageType(source: Source[ByteString, _]): EitherT[Future, FailedToValidateError, MessageType]
+}
+
+@Singleton
+class MessagesXmlParsingServiceImpl @Inject() (implicit materializer: Materializer) extends MessagesXmlParsingService {
+
+  override def validateMessageType(source: Source[ByteString, _]): EitherT[Future, FailedToValidateError, MessageType] =
```

Comment:
You're not validating here, you're extracting the message type, so the name is bad. Equally, you should be using a new left here -- so not `FailedToValidateError` but something else, perhaps `ExtractionError`, and you should create the required cases and typeclass in `ErrorTranslator` to support this.

Reply:
changed method name and added Error Translator